tembo-operator/tests/integration_tests.rs

```diff
@@ -5850,6 +5862,246 @@ CREATE EVENT TRIGGER pgrst_watch
+                                {
+                                    "name": "ferretdb-data",
+                                    "ephemeral": {
+                                        "volumeClaimTemplate": {
+                                            "spec": {
+                                                "accessModes": [
+                                                    "ReadWriteOnce"
+                                                ],
+                                                "resources": {
+                                                    "requests": {
+                                                        "storage": "1Gi"
+                                                    }
+                                                }
+                                            }
+                                        }
+                                    }
+                                }
+                            ],
+                            "volumeMounts": [
+                                {
+                                    "name": "ferretdb-data",
+                                    "mountPath": "/state"
+                                }
+                            ]
+                        },
+                    }
+                ]
+            }
+        });
+
+        let mut cluster_stop = cluster_start.clone();
+        cluster_stop["spec"]["stop"] = serde_json::json!(true);
+
+        // Begin by starting the cluster and validating that it worked.
+        // We need this here as initial iterations of the hibernate code
+        // prevented the cluster from starting.
+
+        let _ = test.set_cluster_def(&cluster_start).await;
+        assert!(status_running(&test.coredbs, &name).await);
+        assert!(pooler_status_running(&test.poolers, &pooler_name).await);
+
+        // Assert there are 4 pods running: postgres, pooler, postgrest and ferretdb
+        let pods: Api<Pod> = Api::namespaced(test.client.clone(), &namespace);
+        let pods_list = pods.list(&Default::default()).await.unwrap();
+        assert_eq!(pods_list.items.len(), 4);
+
+        // Stop the cluster and check to make sure it's not running to ensure
+        // hibernate is doing its job.
+
+        let _ = test.set_cluster_def(&cluster_stop).await;
+        let _ = wait_until_status_not_running(&test.coredbs, &name).await;
+        assert!(status_running(&test.coredbs, &name).await.not());
+        assert!(pooler_status_running(&test.poolers, &pooler_name)
+            .await
+            .not());
+
+        // Assert there are no pods running
+        let pods_list = pods.list(&Default::default()).await.unwrap();
+        assert_eq!(pods_list.items.len(), 0);
+
+        // Assert that IngressRouteTCPs are removed after hibernation
+        // TODO(ianstanton) Typically there should be 0 here, but we're creating one for postgrest
+        //  due to a bug. This should be updated to 0 after the bug is fixed.
```

Comment:
@ChuckHend I pulled out the bug fix we discussed and added some notes for us to update these assertions once that fix is in.